Versioning.md

```diff
@@ -0,0 +1,76 @@
+# Versioning
+
+This document describes the versioning policy for this repository.
+
+## Goals
+
+### API Stability
+
+Once the API for a given signal (spans, logs, metrics, baggage) has been officially released, that API module will function with any SDK that has the same major version, and equal or greater minor or patch version.
+
+For example, libraries that are instrumented with `opentelemetry 1.0.1` will
+function in applications using `opentelemetry 1.11.33` or `opentelemetry
+1.3.4`.
+
+### SDK Stability
+
+Public portions of the SDK (constructors, configuration, end-user interfaces)
+must remain backwards compatible. Internal types are allowed to break.
+
+## Policy
+
+* Releases will follow [SemVer](https://semver.org/).
+* Only single source package containing api and sdk for all singals would be released as part of new GitHub release.
+* New telemetry signals will be introduced behind experimental feature flag usng C macro define.
+  ```
+    #ifdef METRICS_PREVIEW
+         <metrics api/sdk definitions>
+    #endif
+  ```
+
+  As we deliver package in souce form, and the user is responsible to build them for their platform, they have to be
+  aware of these feature flags (documented in CHANGELOG.md), and enable it explicitly through their build system (cmake
+  / bazel or others ) to use the preview feature.
+
+* New signals will be stabilized via a **minor version bump**, and are not allowed to break existing stable interfaces.
+Feature flag would be removed once we have a stable implementation for the signal.
+
+* GitHub releases will be made for all released versions.
+
+## Example Versioning Lifecycle
+
+Purely for illustration purposes, not intended to represent actual releases:
+
+- v0.0.1 release:
+   - `opentelemetry 0.0.1`
+     - Contains experimental impls of trace, resouce ( no feature flag as major version is 0 )
+     - No implementation of logging and metrics available
+   - `opentelemetry-contrib 0.0.1`
```

Comment:
I thought we were to provide two packages:
- `opentelemetry-api` - header only.
- `opentelemetry-sdk` - implementation.

Do we need to cover `opentelemetry-contrib` in this repository? It'd be problematic to consolidate, or decide what is to. be included in `-contrib`. Most likely it would end up being `opentelemetry-${VendorA}` ?

Reply:
Yes, I will change it accordingly. We do add version information in both api and sdk to differentiate their versions.  But honestly I still don't understand the concept of separate packages yet from C++ prospective. If we talk about output of out release management, it would still be single source zip/tar file containing both api and sdk. Hope we are not talking about `vcpkg` here : )

Reply:
Done changes. And removed `-contrib`. I actually felt there should be way for user's of `-contrib` repo to know which version of `-api/sdk` they would need for their exporter to work correctly.